### PR TITLE
Add `RegexIterator` stub.

### PR DESCRIPTION
**Pull request by drupol**

### CODE REVIEW

File: stubs/iterable.stub [hunk 1]
```diff
@@ -298,3 +298,35 @@ class AppendIterator extends IteratorIterator {
...
 }
+
+/**
+ * @template TKey
+ * @template TValue
+ * @template TIterator of Traversable<TKey, TValue>
+ *
+ * @template-extends FilterIterator<TKey, TValue, TIterator>
+ */
+class RegexIterator extends FilterIterator {
```
Comment (ondrejmirtes) [inline stubs/iterable.stub:415]: Needs to be in skipCheckGenericClasses

File: stubs/iterable.stub [hunk 2]
```diff
@@ -298,3 +298,35 @@ class AppendIterator extends IteratorIterator {
...
+    const USE_KEY = 1 ;
+
+    /**
+     * @param Iterator<TKey, TValue> $iterator
+     * @param self::MATCH|self::GET_MATCH|self::ALL_MATCHES|self::SPLIT|self::REPLACE $mode
+     */
+    public function __construct(Iterator $iterator, string $regex, int $mode = self::MATCH, int $flags = 0, int $preg_flags = 0) {}
+
+    /**
+     * @return TValue|null
```
Comment (ondrejmirtes) [inline stubs/iterable.stub]: Please do not add the |null part here.

File: stubs/iterable.stub [hunk 3]
```diff
@@ -298,3 +298,35 @@ class AppendIterator extends IteratorIterator {
...
+     */
+    public function current() {}
+
+    /**
+     * @return TKey|null
```
Comment (ondrejmirtes) [inline stubs/iterable.stub]: dtto

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: stubs/iterable.stub (hunk 2 above)
Comment (drupol) [reply stubs/iterable.stub]: I got rid of the method as we are extending `FilterIterator`.

File: stubs/iterable.stub (hunk 3 above)
Comment (drupol) [reply stubs/iterable.stub]: same here.

File: stubs/iterable.stub (hunk 1 above)
Comment (drupol) [reply stubs/iterable.stub:415]: fixed.

File: stubs/iterable.stub (hunk 2 above)
Comment (drupol) [reply stubs/iterable.stub]: Actually those methods are not defined in `FilterIterator`. Adding them back and remove the `null`.

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**